technico-v16/app/login/page.tsx

```diff
@@ -0,0 +1,111 @@
+"use client";
+
+import { useState } from "react";
+import { useRouter } from "next/navigation";
+import { useAuth } from "../components/AuthContext";
+import { URLS } from "../lib/constants";
+
+interface LoginRequestDto
+{
+    email: string,
+    password: string
+}
+
+interface LoginResponseDto
+{
+ id: string,
+ email: string,
+ isAdmin: boolean
+}
+
+const Login = () => {
+    const [email, setEmail] = useState("");
+    const [password, setPassword] = useState("");
+    const [error, setError] = useState<string | null>(null);
+    const [isLoading, setIsLoading] = useState(false);
+    const router = useRouter();
+    const {fireReload} = useAuth();
+    const [user, setUser] = useState<LoginResponseDto | null>(null);
+    
+    const handleSubmit = async (e: React.FormEvent<HTMLFormElement>) => {
+        e.preventDefault();
+
+            // Fetch Users
+            try {
+              setIsLoading(true);
+              const response = await fetch("https://localhost:7118/api/Users/login", {
+                method: "POST",
+                headers: {
+                    "Content-Type": "application/json"
+                },
+                body: JSON.stringify({email, password})
+              });
+
+              if (!response.ok) {
+                throw new Error(`Error ${response.status}: Failed to fetch user data`);
+              }
+              
+              const userData: LoginResponseDto = await response.json();
+              setUser(userData);
+
+              
+              localStorage.setItem("userId", userData.id)
+              
+              localStorage.setItem("token", userData.isAdmin ? "mock-admin-token" : "mock-user-token");
+
+              console.log("Login successful:", userData);
+
+              // Redirect based on role
+              if (!userData.isAdmin) {
+                router.push("/");
+              }
+              else {
+                router.push("/");
+              }
+
+            } catch (err: any) {
+              setError(err.message || "An error occurred while fetching user data.");
+            } finally {
+              setIsLoading(false);;            
+          }; 
+
+        fireReload();
```

Comment:
Good use of the fireReload function, the guy who thought of it must be a genius :)